### ISSUE TITLE
SEUS PTGI HRR 2.1 cannot render grass sides correctly in reflections and underwater.

### ISSUE THREAD

**Issue by BlueStaggo** [IrisShaders/Iris#1102]
### What happened?

When going underwater, the sides of grass blocks do not render correctly and appear fully green.

### Screenshots

![2021-12-16_15 58 33](https://user-images.githubusercontent.com/71090710/146405682-50177e16-b495-4abd-8b31-08040057ceca.png)
![2021-12-16_15 58 34](https://user-images.githubusercontent.com/71090710/146405696-0945cb60-02e7-4147-bc9c-5b437a49621e.png)

### Relevant log output

```shell
[15:56:48] [Render thread/INFO]: Destroying pipeline OVERWORLD
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "stone_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:stone_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "warped_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:warped_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "crimson_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:crimson_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "polished_blackstone_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:polished_blackstone_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "warped_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:warped_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "crimson_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:crimson_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "oak_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:oak_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "spruce_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:spruce_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "birch_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:birch_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "jungle_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:jungle_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "acacia_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:acacia_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "dark_oak_pressure_plate:power=0":
[15:56:48] [Render thread/WARN]: - The block minecraft:dark_oak_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "stone_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:stone_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "warped_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:warped_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "crimson_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:crimson_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "polished_blackstone_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:polished_blackstone_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "warped_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:warped_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "crimson_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:crimson_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "oak_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:oak_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "spruce_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:spruce_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "birch_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:birch_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "jungle_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:jungle_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "acacia_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:acacia_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/WARN]: Error while parsing the block ID map entry "dark_oak_pressure_plate:power=1":
[15:56:48] [Render thread/WARN]: - The block minecraft:dark_oak_pressure_plate has no property with the name power, ignoring!
[15:56:48] [Render thread/INFO]: Using shaderpack: SEUS PTGI HRR Test 2.1.zip
[15:56:48] [Render thread/INFO]: Creating pipeline OVERWORLD
[15:56:51] [Render thread/WARN]: [Triforce Patcher] gl_FragColor is not supported yet, please use gl_FragData! Assuming that the shaderpack author intended to use gl_FragData[0]...
[15:56:51] [Render thread/WARN]: Program link log for composite5: Fragment info
-------------
0(5700) : warning C7050: "col" might be used before being initialized

[15:56:51] [Render thread/WARN]: [Triforce Patcher] gl_FragColor is not supported yet, please use gl_FragData! Assuming that the shaderpack author intended to use gl_FragData[0]...
[15:56:54] [Render thread/INFO]: Stopping worker threads
[15:56:54] [Render thread/INFO]: Started 6 worker threads
[15:56:54] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_shadow: Geometry info
-------------
0(475) : warning C7050: "zrEuGoMOCd" might be used before being initialized

[15:56:54] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_shadow: Geometry info
-------------
0(475) : warning C7050: "zrEuGoMOCd" might be used before being initialized

[15:56:54] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_gbuffers_terrain: Vertex info
-----------
0(3064) : warning C7050: "tangent" might be used before being initialized
0(3065) : warning C7050: "binormal" might be used before being initialized
0(3054) : warning C7050: "preDownscaleProjPos" might be used before being initialized

[15:56:54] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_gbuffers_terrain_cutout: Vertex info
-----------
0(3064) : warning C7050: "tangent" might be used before being initialized
0(3065) : warning C7050: "binormal" might be used before being initialized
0(3054) : warning C7050: "preDownscaleProjPos" might be used before being initialized

[15:56:54] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_gbuffers_water: Vertex info
-----------
0(531) : warning C7050: "preDownscaleProjPos" might be used before being initialized
0(535) : warning C7050: "tangent" might be used before being initialized
0(536) : warning C7050: "binormal" might be used before being initialized

[15:56:56] [Render thread/INFO]: Destroying pipeline OVERWORLD
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "stone_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:stone_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "warped_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:warped_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "crimson_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:crimson_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "polished_blackstone_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:polished_blackstone_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "warped_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:warped_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "crimson_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:crimson_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "oak_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:oak_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "spruce_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:spruce_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "birch_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:birch_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "jungle_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:jungle_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "acacia_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:acacia_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "dark_oak_pressure_plate:power=0":
[15:56:57] [Render thread/WARN]: - The block minecraft:dark_oak_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "stone_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:stone_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "warped_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:warped_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "crimson_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:crimson_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "polished_blackstone_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:polished_blackstone_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "warped_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:warped_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "crimson_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:crimson_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "oak_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:oak_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "spruce_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:spruce_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "birch_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:birch_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "jungle_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:jungle_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "acacia_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:acacia_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/WARN]: Error while parsing the block ID map entry "dark_oak_pressure_plate:power=1":
[15:56:57] [Render thread/WARN]: - The block minecraft:dark_oak_pressure_plate has no property with the name power, ignoring!
[15:56:57] [Render thread/INFO]: Using shaderpack: SEUS PTGI HRR Test 2.1.zip
[15:56:57] [Render thread/INFO]: Creating pipeline OVERWORLD
[15:56:57] [Render thread/WARN]: [Triforce Patcher] gl_FragColor is not supported yet, please use gl_FragData! Assuming that the shaderpack author intended to use gl_FragData[0]...
[15:56:57] [Render thread/WARN]: Program link log for composite5: Fragment info
-------------
0(5700) : warning C7050: "col" might be used before being initialized

[15:56:57] [Render thread/WARN]: [Triforce Patcher] gl_FragColor is not supported yet, please use gl_FragData! Assuming that the shaderpack author intended to use gl_FragData[0]...
[15:56:57] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_shadow: Geometry info
-------------
0(475) : warning C7050: "zrEuGoMOCd" might be used before being initialized

[15:56:57] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_shadow: Geometry info
-------------
0(475) : warning C7050: "zrEuGoMOCd" might be used before being initialized

[15:56:57] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_gbuffers_terrain: Vertex info
-----------
0(3064) : warning C7050: "tangent" might be used before being initialized
0(3065) : warning C7050: "binormal" might be used before being initialized
0(3054) : warning C7050: "preDownscaleProjPos" might be used before being initialized

[15:56:57] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_gbuffers_terrain_cutout: Vertex info
-----------
0(3064) : warning C7050: "tangent" might be used before being initialized
0(3065) : warning C7050: "binormal" might be used before being initialized
0(3054) : warning C7050: "preDownscaleProjPos" might be used before being initialized

[15:56:57] [Render thread/WARN]: Program link log for sodium:chunk_shader_for_gbuffers_water: Vertex info
-----------
0(531) : warning C7050: "preDownscaleProjPos" might be used before being initialized
0(535) : warning C7050: "tangent" might be used before being initialized
0(536) : warning C7050: "binormal" might be used before being initialized
```

### Minecraft Version

Minecraft 1.18.1

### Iris Version

iris-mc1.18.1-1.1.3.jar

### Sodium Version

sodium-fabric-mc1.18-0.4.0-alpha5+build.9.jar

### Operating System

Windows 11

### What is your GPU?

Nvidia GeForce GTX 1650

### Java Version

Java 17

**Comment by IMS212** [comment]
Are you sure this does not happen on OptiFine 1.18.1?

**Comment by DieuDeGlace** [comment]
This indeed does not happen on optifine 1.18.1. Just tested it out myself.

**Comment by BlueStaggo** [comment]
I can confirm that this does not happen on Optifine 1.18.1 HD H4 Ultra on my GTX 1650.

**Comment by Vir7ualll** [comment]
My issue is a little different as it affects every block reflected and not just grass
![2022-04-23_15 42 29](https://user-images.githubusercontent.com/89110960/164943959-e0e56906-fbf7-4071-9b6b-68c721392815.png)
I use Compliance/Faithful x64 which actually helps show what the issue could be. Looking closer I can tell that the blocks in water and reflections render a corner of the texture for the whole block instead of the whole texture on the whole block properly. Maybe this could help? Here is the pack website if needed: https://faithfulpack.net/ (I use the latest beta so as of writing Beta 4 for x64)

**Comment by Vir7ualll** [comment]
I updated to Iris 1.2.3 as well as to the latest Indium and Fabric API 0.51.1 and the issue is still prevalent. I tried messing with the other mods I have (removing and re adding etc) and the issue is always there even when the only things I have are Fabric API, Sodium, and Iris. That will be all for my updates unless more is requested.

**Comment by coderbot16** [comment]
Unable to reproduce with Iris 1.2.4 on 1.18.2 using SEUS PTGI HRR 2.1. SEUS PTGI HRR 3 is not currently supported and support for it is being tracked in a separate issue.

![2022-04-29_17 14 23](https://user-images.githubusercontent.com/22420959/166082841-72c91626-7ce9-4e0b-b86e-c6d6d6332a42.png)
![2022-04-29_17 14 28](https://user-images.githubusercontent.com/22420959/166082883-c2ce2466-1041-491c-b264-914abd60905c.png)
![2022-04-29_17 14 44](https://user-images.githubusercontent.com/22420959/166082890-7dff3995-a930-4afe-9a6e-b7f94cc4d2b4.png)

**Comment by coderbot16** [comment]
While there are some cases where this can happen, the behavior is the same on OptiFine:
![image](https://user-images.githubusercontent.com/22420959/166082975-2ac806cf-9645-46fe-9b9c-571bf7c1d136.png)

**Comment by Vir7ualll** [comment]
I know this issue is closed but for me it still persists, Iris 1.2.4
Glass reflections are negligible to bring up, they have the same zoom-into-the-texture-corner issue on Iris but it is less broken than glass reflections are on optifine, with them straight up placing the glass lines on every texture there. The textures through water are still broken just the same
![2022-05-11_20 54 06](https://user-images.githubusercontent.com/89110960/167971049-e55c3fe6-221d-417f-89e0-8ebd8bb3c2c6.png)
This is the Iris screenshot, specs are still the same as what I gave earlier
![2022-05-11_20 48 24](https://user-images.githubusercontent.com/89110960/167971088-e37e2f35-c1db-4a6e-a55c-4254ed9899b2.png)
This is the Optifine screenshot, with the textures rendering fully properly.
All of this is using HRR 2.1

**Comment by Vir7ualll** [comment]
I know this issue is closed but for me it still persists, Iris 1.2.4
Glass reflections are negligible to bring up, they have the same zoom-into-the-texture-corner issue on Iris but it is less broken than glass reflections are on optifine, with them straight up placing the glass lines on every texture there. The textures through water are still broken just the same
![2022-05-11_20 54 06](https://user-images.githubusercontent.com/89110960/167971049-e55c3fe6-221d-417f-89e0-8ebd8bb3c2c6.png)
This is the Iris screenshot, specs are still the same as what I gave earlier
![2022-05-11_20 48 24](https://user-images.githubusercontent.com/89110960/167971088-e37e2f35-c1db-4a6e-a55c-4254ed9899b2.png)
This is the Optifine screenshot, with the textures rendering fully properly.
All of this is using HRR 2.1
